### PR TITLE
feat(front): add inline images

### PR DESCRIPTION
**Pull request by germanurrus**
Screen Shot 2022-04-26 at 22 16 37

Screen Shot 2022-04-26 at 22 16 48

### CODE REVIEW

File: frontend/packages/client/src/components/UploadImageModal.js [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+    onUploaded,
+    loading,
+    error,
+    imageKey,
+    onUploadStared,
+    deleteImage,
+    uploadFile,
+  ]);
+
+  const fileName = image?.name ?? image.file.name;
```
Comment (CJamesKeller) [inline frontend/packages/client/src/components/UploadImageModal.js:51]: Not a necessary change, but if we are checking whether `image` exists here in the first term, I would think the check should be applied in the second term, as well.

File: frontend/packages/client/src/components/UploadImageModal.js [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+import { Upload, Bin } from "../components/Svg";
+import { Loader } from "../components";
+
+const IMAGE_STATUS = {
+  notStarted: "not-started",
+  uploading: "uploading",
+  uploaded: "uploaded",
+};
+
+function ImageUploader({
```
Comment (CJamesKeller) [inline frontend/packages/client/src/components/UploadImageModal.js:13]: In general, I would favor splitting components into separate files and using folders to group them by functionality/larger components, with the top-level `components` folder being mainly for those which are shared across multiple areas. What do you think?

**Review by CJamesKeller — APPROVED** [review]
Looks good! A lot of functionality here -- Thank you for tackling this!

**Comment by germanurrus** [comment]
@CJamesKeller I'm doing one last update on this PR: image uploads should be restricted to one and adding a caption below the image

File: frontend/packages/client/src/components/UploadImageModal.js (hunk 2 above)
Comment (germanurrus) [reply frontend/packages/client/src/components/UploadImageModal.js:13]: yes, I agree. We should take this approach...perhaps there might be some refactor to be done to achieve this on all the repo but I think it's convenient effort

**Comment by germanurrus** [comment]
@CJamesKeller @dbslone updates done here. It's Ready for review: I'll create a second PR to adjusting the style for the caption text added at the bottom of the image.

**Review by dbslone — APPROVED** [review]
LGTM

**Pull request merged**